gcloud/_helpers.py

```diff
@@ -238,3 +241,35 @@ def __init__(self, project=None, implicit=False):
 
 
 _DEFAULTS = _DefaultsContainer(implicit=True)
+
+
+class _ClientProxy(object):
```

Comment:
Yes I understand the use of proxy after seeing `_Topic` etc. below. However, IMO it is over-engineering.

As I mentioned, I'd like to see a smaller spec for this PR and iterate from there.